### PR TITLE
Fix Mailer DSNs for 4.4

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| License       | MIT
| Doc issue/PR  | symfony/symfony-docs#...

### CODE REVIEW

**Comment by fabpot** [comment]
This pull request updates the DSNs and also removes the intermediary env vars for following reasons:

* Having env vars embedded into other ones does not work well with Dotenv;
* Having less env vars is "better"
* Having the user/pass/key/id embedded into the DSN makes it easier to understand that they need to be URL encoded.

File: symfony/google-mailer/4.4/manifest.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "env": {
+        "#1": "Gmail SHOULD NOT be used on production, use it in development only.",
```
Comment (Koc) [inline symfony/google-mailer/4.4/manifest.json]: Should comment to be prefixed with `#`? Also why Gmail is not recommended for prod?

Comment (stof) [reply symfony/google-mailer/4.4/manifest.json]: it is prefixed (in the name of the env var, which is what goes at the beginning of the line)

Comment (fabpot) [reply symfony/google-mailer/4.4/manifest.json]: Gmail is not meant to be used to send emails from an application. Accounts are personal. So, it works for testing on a dev machine, but not for prod.

File: symfony/sendgrid-mailer/4.4/manifest.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "env": {
+        "#1": "MAILER_DSN=sendgrid://KEY@default"
```
Comment (bocharsky-bw) [inline symfony/sendgrid-mailer/4.4/manifest.json]: Shouldn't it be `MAILER_DSN=sendgrid+smtp://KEY@default` to be consistent with 4.3 recipe?

Comment (fabpot) [reply symfony/sendgrid-mailer/4.4/manifest.json]: `sendgrid` is the equivalent of `sendgrid+smtp`. The default is now automatically the "best" transport possible.

Comment (bocharsky-bw) [reply symfony/sendgrid-mailer/4.4/manifest.json]: I got it, thank you for the clarification!

**Pull request merged**